src/adapters/collateralized_simple_interest_loan_adapter.ts

```diff
@@ -113,3 +134,214 @@ export class CollateralizedLoanTerms {
+            // omit the simple interest parameters that will be packed into the `termsContractParameters`.
+            "principalTokenSymbol",
+            "interestRate",
+            "amortizationUnit",
+            "termLength",
+            // omit the collateralized parameters that will be packed into the `termsContractParameters`.
+            "collateralTokenSymbol",
+            "collateralAmount",
+            "gracePeriodInDays",
+        ]);
+
+        const packedSimpleInterestParams = this.simpleInterestLoanTerms.packParameters({
+            principalTokenIndex,
+            principalAmount,
+            interestRate,
+            amortizationUnit,
+            termLength,
+        });
+
+        const packedCollateralizedParams = this.collateralizedLoanTerms.packParameters({
+            collateralTokenIndex,
+            collateralAmount,
+            gracePeriodInDays,
+        });
+
+        // Our final output is the perfect union of the packed simple interest params and the packed
+        // collateralized params.
+        const packedParams =
+            packedSimpleInterestParams.substr(0, 39) + packedCollateralizedParams.substr(39, 27);
+
+        debtOrder = {
+            ...debtOrder,
+            principalToken: principalToken.address,
+            termsContract: collateralizedSimpleInterestTermsContract.address,
+            termsContractParameters: packedParams,
+        };
+
+        return DebtOrder.applyNetworkDefaults(debtOrder, this.contractsAPI);
+    }
+
+    public async fromDebtOrder(
+        debtOrder: DebtOrder.Instance,
+    ): Promise<CollateralizedSimpleInterestLoanOrder> {
+        this.assert.schema.debtOrderWithTermsSpecified("debtOrder", debtOrder);
+
+        const {
+            principalTokenIndex,
+            principalAmount,
+            interestRate,
+            termLength,
+            amortizationUnit,
+        } = this.simpleInterestLoanTerms.unpackParameters(debtOrder.termsContractParameters);
+
+        const {
+            collateralTokenIndex,
+            collateralAmount,
+            gracePeriodInDays,
+        } = this.collateralizedLoanTerms.unpackParameters(debtOrder.termsContractParameters);
+
+        const principalTokenSymbol = await this.contractsAPI.getTokenSymbolByIndexAsync(
+            principalTokenIndex,
+        );
+
+        const collateralTokenSymbol = await this.contractsAPI.getTokenSymbolByIndexAsync(
+            collateralTokenIndex,
+        );
+
+        const collateralTokenAddress = await this.contractsAPI.getTokenAddressBySymbolAsync(
+            collateralTokenSymbol,
+        );
+
+        // Assert that the principal token corresponds to symbol we've unpacked.
+        this.assertTokenCorrespondsToSymbol(debtOrder.principalToken, principalTokenSymbol);
+
+        // Assert that the collateral token address corresponds to symbol we've unpacked.
+        this.assertTokenCorrespondsToSymbol(collateralTokenAddress, collateralTokenSymbol);
```

Comment:
Is this check really necessary given that `collateralTokenAddress` isn't user inputted -- i.e. we just pulled the address via our own APIs.

Reply:
Agreed